exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -330,23 +336,56 @@ func (prwe *PrwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("User-Agent", prwe.userAgentHeader)
 
-	resp, err := prwe.client.Do(req)
-	if err != nil {
-		return consumererror.Permanent(err)
+	return prwe.makeReqWithRetries(req)
+}
+
+func (prwe *PrwExporter) makeReqWithRetries(req *http.Request) error {
+	backoff := prwe.cfg.RemoteWriteQueue.MinBackoff
+	if backoff == 0 {
+		// Should never happen, but setting it to be safe.
+		backoff = defaultRetryMinBackoff
+	}
+	maxBackoff := prwe.cfg.RemoteWriteQueue.MaxBackoff
+	if maxBackoff == 0 {
+		// Should never happen, but setting it to be safe.
+		backoff = defaultRetryMaxBackoff
 	}
-	defer resp.Body.Close()
 
+	var lastErr error
+	for ; backoff <= maxBackoff; backoff *= 2 {
+		retry, err := prwe.makeReq(req)
+		if err == nil {
+			return nil
+		}
+		if !retry {
+			return err // permanent error
+		}
+
+		lastErr = err // transient error
+		time.Sleep(backoff)
```

Comment:
I think random jitter would be also desirable. Consider using "github.com/cenkalti/backoff/v4" which provides a nice exponential backoff algo out of the box (we already use it in queued_retry helper).